server/web/filter/ratelimit/token_bucket.go

```diff
@@ -0,0 +1,76 @@
+package ratelimit
+
+import (
+	"sync"
+	"time"
+)
+
+type tokenBucket struct {
+	sync.RWMutex
+	remaining   uint
+	capacity    uint
+	lastCheckAt time.Time
+	rate        time.Duration
+}
+
+// NewTokenBucket return an bucket that implements token bucket
+func NewTokenBucket(opts []BucketOption) Bucket {
```

Comment:
Using `opts ... BucketOption`. In general, when we use `option design pattern`, we always use `opts...XXXXOpt`